src/scraper/http.cpp

```diff
@@ -1,13 +1,27 @@
 #include "http.h"
 #include "tinyformat.h"
+#include "util.h"
 
 #include <curl/curl.h>
 #include <stdio.h>
 #include <string.h>
 #include <memory>
+#include <boost/thread.hpp>
 
-// Only for troubleshooting. This should be removed before production.
-#include "util.h"
+#if LIBCURL_VERSION_NUM >= 0x073d00
+/* In libcurl 7.61.0, support was added for extracting the time in plain
+   microseconds. Lets make this so we can allow a user using own depends
+   for compile. */
+#define timetype curl_off_t
+#define timeopt CURLINFO_TOTAL_TIME_T
+#define progressinterval 3000000
+#else
+#define timetype double
+#define timeopt CURLINFO_TOTAL_TIME
+#define progressinterval 1
+#endif
+
+struct_SnapshotStatus DownloadStatus;
```

Comment:
If a user starts the Qt wallet with `-snapshotdownload` and then tries again to install the snapshot using the menu item, the snapshot process fails because this global state is not reset after the first round finishes.

Reply:
two things I think we should do here. one if the argument exists then we should not allow the gui menu option to be enabled since it would of had to be successful to have the wallet load. and the status should be cleared after use.

Reply:
Even though it does the same thing as the argument, it might be confusing for the user if we hide the menu item, and a wallet could potentially stay up for weeks before a user looks for the menu item again. I think we can skip the extra code needed to show and hide it.

Reply:
solved this. in upgrade constructor it'll clear those values.